nestkernel/connector_base.h

```diff
@@ -352,12 +506,51 @@ class Connector< 1, ConnectionT > : public vector_like< ConnectionT >
       return 0;
   }
 
+  size_t
+  get_num_connections( size_t target_gid, size_t thrd, synindex syn_id )
+  {
+    size_t num_connections = 0;
+    if ( syn_id == get_syn_id() )
+    {
+      if ( C_[ 0 ].get_target( thrd )->get_gid() == target_gid )
+      {
+        num_connections = 1;
+      }
+    }
+    return num_connections;
+  }
+
   Connector< 2, ConnectionT >&
   push_back( const ConnectionT& c )
   {
     return *suicide_and_resurrect< Connector< 2, ConnectionT > >( this, c );
   }
 
+  ConnectorBase& erase( size_t )
+  {
+    // Should destroy the Connector
+    suicide< Connector< 1, ConnectionT > >( this );
+    ConnectorBase* p = 0;
+    return *p;
+  }
+
+  size_t
+  size()
+  {
+    return 1;
+  }
+
+  ConnectionT&
+  at( size_t i )
+  {
+
+    if ( i != 0 )
+    {
+      throw std::out_of_range( "Invalid attempt to access a connection at an out of range index" );
```

Comment:
Can you please rephrase? Something like "Invalid attempt to access connection: index yyy out of range." For adding the number to the string you can use the String::compose function used in many places of NEST.